nixos/modules/services/misc/mediatomb.nix

```diff
@@ -6,37 +6,97 @@ let
 
   gid = config.ids.gids.mediatomb;
   cfg = config.services.mediatomb;
+  name = cfg.package.pname;
+  pkg = cfg.package;
+  optionYesNo = option: if option then "yes" else "no";
+  # configuration on media directory
+  mediaDirectory = {
+    options = {
+      path = mkOption {
+        type = types.str;
+        description = ''
+          Absolute directory path to the media directory to index
+        '';
+      };
+      recursive = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether the indexation must take place recursively or not";
+      };
+      hidden-files = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to index the hidden files or not";
+      };
+    };
+  };
+  toMediaDirectory = d: "<directory location=\"${d.path}\" mode=\"inotify\" recursive=\"${optionYesNo d.recursive}\" hidden-files=\"${optionYesNo d.hidden-files}\" />\n";
 
-  mtConf = pkgs.writeText "config.xml" ''
-  <?xml version="1.0" encoding="UTF-8"?>
-  <config version="2" xmlns="http://mediatomb.cc/config/2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://mediatomb.cc/config/2 http://mediatomb.cc/config/2.xsd">
+  transcodingConfig = if cfg.transcoding then with pkgs; ''
```

Comment:
It has been too many years since I have used `mediatomb` to recall all the details, but would it be better for us to do xml substitutions on the configuration file provided by upstream instead of generating our own? I'm not sure what the right answer is here but would like to hear your thoughts.

Reply:
Well, as far as I could tell, they don't provide one. The new gerbera cli
provide the means to generate one though but that hardcodes values... (thus not
working for nixos's case).

So I think continuing that way is fine. The benefit from doing this ourselves
allows to be consistent between what the user declares and the generated
configuration.

We can also benefit from nix's laziness (like I did when fixing the transcoding
option, if we don't activate it here, the default, we don't actually pull vlc
or ffmpeg ;)

The only drawback I see continuing that way is well maintenance here ¯\_(ツ)_/¯ (which should not that much of a problem if we continue maintaining this with tests (maybe ;)

Reply:
Note that, as another example, I also leveraged nix's laziness here to stop completely the configuration file generation if the user enables customCfg (which as far as I understood it was not the case before ;).